src/main/java/com/uid2/admin/auth/AdminKeyset.java

```diff
@@ -0,0 +1,32 @@
+package com.uid2.admin.auth;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.uid2.admin.model.ClientType;
+import com.uid2.shared.auth.Keyset;
+
+import java.util.Set;
+
+public class AdminKeyset extends Keyset {
```

Comment:
I had a discussion with Andrei about this and he suggested this approach. I see it as two reasons why
1. Only admin really needs to know about sharing types, so this prevents someone from misusing it in shared
2. This makes the changes way easier because now only admin needs to change for this, also any future logic we want to add to keysets can be added in a similar fashion. 

Lets talk about composition vs inheritance on a call? I think I might agree